### ISSUE TITLE
ELI5 instructions

### ISSUE THREAD

**Issue by magick93** [tnozicka/openshift-acme#32]
Hello

I have installed the controller (appuio/ansible-role-openshift-acme)

On the Readme it says:

>Enabling ACME certificates for your object

>metadata:
> annotations:
>   kubernetes.io/tls-acme: "true"

I added this to my openshift route yml and service yml, but I still get the *Your connection is not private* error.

Specifically, where should I add this annotation?

**Comment by tnozicka** [comment]
@magick93 the annotation should be added to your Route. Check the controller logs (using `oc logs `) to see what's happening.

**Comment by tnozicka** [comment]
also check the certificate either using curl or e.g. chrome and check the issuer. You might have deployed against Let's Encrypt staging instead of life

**Comment by magick93** [comment]
Thanks.
Does the controller need to be installed in the same project as the route?

**Comment by tnozicka** [comment]
you have probably deployed it cluster-wide, but that's just a guess since you are not using our predefined deplyoments in deploy/ folder nor you have provided the config/values

**Comment by magick93** [comment]
How would I tell if I deployed it custer-wide?

**Comment by tnozicka** [comment]
default is cluster-wide, unless you restrict it to watch certain namespaces with `-w` flag or `OPENSHIFT_ACME_WATCH_NAMESPACE` variable

**Comment by magick93** [comment]
Ok thanks.

In Chrome, I get the following. This is from creating wildfly app. In the logs it says that this cert was created successfully - although there were a lot of errors prior to this.:

```
Attackers might be trying to steal your information from wildlfy-test.apps.mydomain.com (for example, passwords, messages, or credit cards). Learn more
NET::ERR_CERT_AUTHORITY_INVALID
```

The logs are at https://gist.github.com/magick93/a2909abf57963f74b7c8992958f62340

**Comment by tnozicka** [comment]
use the Security tab in chrome; click on view certificate and check the issuer

**Comment by tnozicka** [comment]
You have deployed staging! https://gist.github.com/magick93/a2909abf57963f74b7c8992958f62340#file-acmecontroller-log-L2

**Comment by tnozicka** [comment]
see https://github.com/tnozicka/openshift-acme/blob/master/deploy/deployment-letsencrypt-live.yaml#L24

**Comment by magick93** [comment]
Ok. Forgive my ignorance. What is staging and production used for? What is the difference?

**Comment by tnozicka** [comment]
staging is for testing and signs certificate using other LE CA that is not trusted in your system but has higher limits so you won't block yourself while trying it out.

https://letsencrypt.org/docs/staging-environment/

**Comment by magick93** [comment]
Thanks. Ok, so I changed it to use the production url, and tried again.

Now I get the following error in the controller:

`2017-09-11T12:13:30.628036535Z  ERROR dbcertentry.go:79 [domain: wfsecuretest2-test.apps.mydomain.com, error: 403 urn:acme:error:unauthorized: No registration exists matching provided key]`

Maybe I missed it on the LetsEncrypt website, but I did not see anywhere to register and get a key.

Where do I set the key? And do I need to register with LetsEncrypt?

**Comment by magick93** [comment]
Ok, I think I resolved the previous error by running:

`oc delete secret acme-account`

**Comment by tnozicka** [comment]
yes, that's it. we don't support switching the accounts without manual intervention; at least yet.

And you getting this wrong for the first time is actually our fault since documentation doesn't state that clearly. I will keep that in mind when I update it but it the meantime PRs are always welcome.

**Comment by magick93** [comment]
> PRs are always welcome.
Then a PR you shall have!

I'll add some beginner guides to the readme.

**Comment by tnozicka** [comment]
@magick93 thx :)
